ansible/roles/test/files/ptftests/py3/advanced-reboot.py

```diff
@@ -1985,31 +1986,56 @@ def examine_flow(self, filename=None):
                     prev_time = received_time
                     continue
                 if received_payload - prev_payload > 1:
-                    # Packets in a row are missing, a disruption.
+                    if received_payload not in sent_packets:
+                        self.log("Ignoring received packet with payload {}, as it was not sent".format(
+                            received_payload))
+                        received_but_not_sent_packets.add(received_payload)
+                        continue
```

Comment:
This will miss the failure case - when there is a received packet drop and also a sent packet drop. We may not care if sent packets are not captured in pcap. But any drop in received packet should be reported.

Do not fail immediately upon noticing a received packet drop. We want to measure all received packet drops before concluding a failed state.

Reply:
For the branch highlighted here, it handles the case where a received packet was present but no corresponding send packet in which case it logs it out, takes note of the received but not sent packet and carries on processing.

Further below, in the try-except while-loop it determines if packet IDs were missed for both send and receive from the pcaps and if so tracks them and continues processing.

Once all packets have been processed, if there were any that were missed from both send and received then these are logged out as a failure in the test will fail.